pkg/cloudprovider/provider/vsphere/client.go

```diff
@@ -50,9 +49,7 @@ func NewSession(ctx context.Context, config *Config) (*Session, error) {
 	// creating the govmoni Client in roundabout way because we need to set the proper CA bundle: reference https://github.com/vmware/govmomi/issues/1200
 	soapClient := soap.NewClient(clientURL, config.AllowInsecure)
 	// set our CA bundle
-	soapClient.DefaultTransport().TLSClientConfig = &tls.Config{
-		RootCAs: util.CABundle,
-	}
+	soapClient.DefaultTransport().TLSClientConfig.RootCAs = util.CABundle
```

Comment:
Are we sure `soapClient.DefaultTransport().TLSClientConfig` is never nil?

Reply:
It gets initialized in soap.NewClient():

`c.t.TLSClientConfig = &tls.Config{InsecureSkipVerify: c.k}`